service/history/workflow/context.go

```diff
@@ -210,6 +210,7 @@ func (c *ContextImpl) Clear() {
 	metrics.WorkflowContextCleared.With(c.metricsHandler).Record(1)
 	if c.MutableState != nil {
 		c.MutableState.GetQueryRegistry().Clear()
+		c.MutableState.RemoveSpeculativeWorkflowTaskTimeoutTask()
```

Comment:
This is extra protection to remove timeout task from memory queue, because WFT is gone anyway, and there is nothing to timeout.